polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisMetaStoreManagerImpl.java

```diff
@@ -1502,6 +1506,26 @@ public Map<String, String> deserializeProperties(PolarisCallContext callCtx, Str
       }
     }
 
+    // Check that cleanup is enabled, if it is set:
+    if (catalogPath != null && cleanup) {
+      boolean dropWithPurgeEnabled =
```

Comment:
On further thought we should probably pull this check into `BasePolarisCatalog` so that the Persistence layer minimizes its exposure to "application layer semantics" that are specific to entity types. I know it's already leaking a bit due to namespace/catalog "not empty" checking, but that can be refactored into "existence of entities with parentId == X".

Since the BasePolarisCatalog ultimately defines the mechanics of what a "DROP_TABLE_WITH_PURGE" means, it'll be best to consolidate the check at that layer.